### ISSUE TITLE
How to train the discriminator network?

### ISSUE THREAD

**Issue by hailuoS** [hfslyc/AdvSemiSeg#1]
In this paper, the discriminator network takes a probability map of size H × W × C and outputs a confidence map of size H × W × 1.  I don't understand how to train this discriminator network, what are the labels during the training phase ? As my understanding, when the probability map comes from the ground truth, the label is a map of size H × W × 1 and the values in the map are all 1, in contrast, when the probability map comes from the segmentation network, the label is a map of size H × W × 1 and the values in the map are all 0. Am I right? Hope to get your answer. Thanks so much

**Comment by hfslyc** [comment]
Yes, you are right. It's like training a GAN with spatial ground truths. Please refer to https://github.com/hfslyc/AdvSemiSeg/blob/master/train.py#L380-L412 for how to train the discriminator network.

**Comment by hailuoS** [comment]
Thanks for your answer! Now I have a new problem, I have run the code, but during the training phase, the value of loss_semi was always 0 (after training for 5000 iterations). I don't know why. After training, I ran the evaluate_voc.py and got the meanIOU: 0.641848480316, this value is lower than the result which I evaluated on the given pre-trained model.
